lib/Dialect/FIRRTL/Transforms/BlackboxMemory.cpp

```diff
@@ -45,10 +46,10 @@ llvm::hash_code computeHash(MemOp op) {
     hash = llvm::hash_combine(hash, resultTypes.front());
     break;
   default:
-    // Use the type buffer as the hash, as we can guarantee it is the same for
+    // Use the type range as the hash, as we can guarantee it is the same for
```

Comment:
This change seemed like the correct thing (a `TypeRange` is hashable and comparable, and I think the upstream change did `ArrayRef<Type>` -> `TypeRange` in a similar way). @youngar I think you own this. Do you mind checking here? The changes in this file were the only thing I felt might not be totally mechanical. Is this covered by the existing tests?

Reply:
Sorry for the slow response, I missed this question.  You can replace the whole case with the hash of result types/

```diff
+ hash = llvm::hash_combine(hash, op->getResultTypes());
-  // Result Types
-  ArrayRef<Type> resultTypes = op->getResultTypes();
-  switch (resultTypes.size()) {
-  case 0:
-    // We don't need to add anything to the hash.
-    break;
-  case 1:
-    // Add in the result type.
-    hash = llvm::hash_combine(hash, resultTypes.front());
-    break;
-  default:
-    // Use the type buffer as the hash, as we can guarantee it is the same for
-    // any given range of result types. This takes advantage of the fact the
-    // result types >1 are stored in a TupleType and uniqued.
-    hash = llvm::hash_combine(hash, resultTypes.data());
-    break;
-  }
-  return hash
```